### ISSUE TITLE
Raise an error if negative value passed as input parameter

### ISSUE THREAD

**Issue by chris-greening** [chris-greening/spyrograph#92]
**Is your feature request related to a problem? Please describe.**
If a negative value is passed as an argument for one or more of the input parameters raise an error warning the user to pass an argument > 0 

i.e.

```python
import numpy as np
from spyrograph import Hypotrochoid

hypotrochoid = Hypotrochoid(R=100, r=-20, d=25, thetas=np.arange(0,2*np.pi, .1))
```

would raise an error like 

"ValueError: Negative value is an invalid input parameter. Please pass values greater than zero"

**Comment by SahilDave04** [comment]
You want it show error message when Negative value is passed as input.

**Comment by chris-greening** [comment]
Heyo @SahilDave04 !! 

Yes - currently the library accepts any input but the inputs are supposed to define the radii and distances from circles so having a negative or zero radius doesn't make sense in this context

I figure a simple check on `__init__` if one or more input args <= 0 then raise `ValueError`

**Comment by SahilDave04** [comment]
else we can just check it when the user is giving input with just an if else statement

**Comment by chris-greening** [comment]
Correct! Are you interested in this issue? I'll assign you and then give a quick brief so you know where to get started!

**Comment by SahilDave04** [comment]
Sure I'd like to try.

**Comment by SahilDave04** [comment]
Can you please direct me to the main file. 
Sorry I'm a newbie.

**Comment by chris-greening** [comment]
Awesome thanks so much for your interest!! :D 

So all of the objects I listed above inherit from an abstract base class called [_Trochoid](https://github.com/chris-greening/spyrograph/blob/main/spyrograph/_trochoid.py#L26) that defines most of the behavior

For this issue you would pretty much just have to drop into this class' `__init__` and place an if check that checks if `self.r`, `self.R`, or `self.d` are negative and then raise the error (right around [here](https://github.com/chris-greening/spyrograph/blob/main/spyrograph/_trochoid.py#L70) is a good spot)

**Comment by chris-greening** [comment]
Don't be sorry at all - I'm here to help! :D Any questions at all please reach out

**Comment by SahilDave04** [comment]
I will.

**Comment by SahilDave04** [comment]
Do you not want the values to be Zero as well?

**Comment by chris-greening** [comment]
Good catch! I actually flagged this in a separate issue #91 - feel free to tackle both at the same time!

**Comment by SahilDave04** [comment]
Do you want it to specifically mention if 0 is passed?

**Comment by chris-greening** [comment]
Nah it can be in the same `if` check that you have in #93, just change < to <= so it'll trigger on zero as well

**Comment by chris-greening** [comment]
Solved with #94
